modules/apps/29-fee/fee_test.go

```diff
@@ -55,6 +64,49 @@ func (suite *FeeTestSuite) CreateMockPacket() channeltypes.Packet {
 	)
 }
 
+// RemoveFeeMiddleware removes:
+// - Fee middleware from transfer module
+// - Fee middleware from icahost submodule
+// - Fee middleware from icacontroller submodule
+// - The entire mock-fee module
+//
+// It does this by overriding the IBC router with a new router.
+func RemoveFeeMiddleware(chain *ibctesting.TestChain) {
+	channelKeeper := chain.GetSimApp().IBCKeeper.ChannelKeeper
+
+	// Unseal the IBC router by force
+	r := reflect.ValueOf(chain.GetSimApp().IBCKeeper.PortKeeper.Router).Elem()
+
+	f := r.FieldByName("sealed")
+	if !f.IsValid() {
+		panic("field not found")
+	}
+	if !f.CanSet() { // Check if we can set the field
+		f = reflect.NewAt(f.Type(), unsafe.Pointer(f.UnsafeAddr())).Elem()
+	}
+
+	f.SetBool(false) // Use reflection to bypass unexported status
```

Comment:
Can't we just `chain.GetSimApp().GetIBCKeeper().PortKeeper.Router = nil` here? since you create a new router below. The check in core `SetRouter` is logical AND on `router == nil && sealed`

Reply:
Ah, good idea.

Reply:
done